### PR TITLE
WIP - Test against multiple framework versions

### PR DESCRIPTION
**Pull request by truenorth**
Continuing the work from #118 :tada:

### CODE REVIEW

**Comment by truenorth** [comment]
This looks like it will require more thought, since this addon supports ember versions older than 1.10 (and it would be nice to keep it that way)

**Comment by cibernox** [comment]
@truenorth Ember try doesn't support older embers?

**Comment by truenorth** [comment]
ember-try doesn't aim to support versions older than 1.10, since it would involve dealing with differences between template compilers and handlebars 1.3, 2.0, and 3.0. Since ember-try is the typical solution for cross-framework-version testing, ... :hurtrealbad: 

https://github.com/kategengler/ember-try#limitations

> Can only change versions for Ember 1.10+ due to template compiler changes that this addon does not attempt to handle.

**Comment by rwjblue** [comment]
@truenorth - I do not think that ember-try is opposed to adding support for swapping NPM packages (which is the current limitation), rather it implemented `bower` support first as it was more straightforward.  In the bower-less future of Ember CLI ember-try will absolutely need to support NPM packages.

**Comment by truenorth** [comment]
Agree @rwjblue. the "requiring more thought" I was referring to in my above comment was mainly referring to some badly needed enhancements/fixes to ember-try.

File: .travis.yml [hunk 1]
```diff
@@ -2,13 +2,29 @@
 language: node_js
 node_js:
   - "0.12"
-  - "iojs"
```
Comment (truenorth) [inline .travis.yml]: No need for us to test against both Node and iojs

File: .travis.yml [hunk 2]
```diff
@@ -2,13 +2,29 @@
...
+  - EMBER_TRY_SCENARIO=ember-canary
+
+matrix:
+  fast_finish: true
+  allow_failures:
+    - env: EMBER_TRY_SCENARIO=ember-1.11
+    - env: EMBER_TRY_SCENARIO=ember-1.12
+    - env: EMBER_TRY_SCENARIO=ember-release
+    - env: EMBER_TRY_SCENARIO=ember-beta
+    - env: EMBER_TRY_SCENARIO=ember-canary
```
Comment (truenorth) [inline .travis.yml]: As we get tests passing for various framework versions, we can disallow failures on those versions

**Comment by truenorth** [comment]
Starting with a straight-forward Ember >= 1.10.0 implementation. I'll backfill Ember < 1.10.0 at some point in the future, after npm support lands in ember-try

**Pull request merged**